voltixApp/VoltixApp/Views/Keygen/PeerDiscoveryView.swift

```diff
@@ -194,12 +198,24 @@ struct PeerDiscoveryView: View {
             .foregroundColor(.red)
     }
     
+    var vaultDetail: some View {
+        Text(viewModel.vaultDetail)
+            .font(.body15MenloBold)
+            .multilineTextAlignment(.center)
+    }
     private func handleSelection(_ peer: String) {
         if viewModel.selections.contains(peer) {
-            viewModel.selections.remove(peer)
+            if peer != viewModel.localPartyID {
+                viewModel.selections.remove(peer)
+            }
         } else {
             viewModel.selections.insert(peer)
         }
+        let totalSigners = viewModel.selections.count
+        if totalSigners >= 3 {
+            let threshold = Int(ceil(Double(totalSigners) * 2.0 / 3.0))
```

Comment:
what does it do?

Reply:
it calculate the threshold , number of signers to do a keysign